text/0000-stable-rustdoc-urls.md

```diff
@@ -0,0 +1,280 @@
+[55160]: https://github.com/rust-lang/rust/issues/55160
+[55160-blog]: https://github.com/rust-lang/rust/issues/55160#issuecomment-680751534
+[intra-doc links]: https://github.com/rust-lang/rfcs/blob/master/text/1946-intra-rustdoc-links.md
+[rust-lang/docs.rs#1055]: https://github.com/rust-lang/docs.rs/issues/1055
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+Rustdoc will make the following changes to URL structure:
+
+- Item pages will be dependent only on the namespace, not the type of the item.
+
+	Consider the struct `std::process::Command`.
+	Currently, the URL for it looks like `std/process/struct.Command.html`.
+	This RFC proposes to change the URL to `std/process/Command.t.html`.
+	Pages named `kind.name.html` would still be generated (to avoid breaking existing links),
+	but would immediately redirect to the new URL.
+
+- Re-exports will generate a page pointing to the canonical version of the documentation.
+
+	Consider the following Rust code:
+	
+	```rust
+	pub struct Foo;
+	```
+	
+	Rustdoc currently generates a page for this at `struct.Foo.html`.
+	Now, consider what happens when you move the struct to a different module and re-export it
+	(which is a semver-compatible change):
+
+	```rust
+	pub mod foo { pub struct Foo; }
+	pub use foo::Foo;
+	```
+
+	This generates a page at `foo/struct.Foo.html`, but _not_ at `struct.Foo.html`.
+	After this change, rustdoc will generate a page at the top level which redirects
+	to the version nested in the module.
+
+<!--
+Explain the proposal as if it was already included in the language and you were teaching it to another Rust programmer. That generally means:
+
+- Introducing new named concepts.
+- Explaining the feature largely in terms of examples.
+- Explaining how Rust programmers should *think* about the feature, and how it should impact the way they use Rust. It should explain the impact as concretely as possible.
+- If applicable, provide sample error messages, deprecation warnings, or migration guidance.
+- If applicable, describe the differences between teaching this to existing Rust programmers and new Rust programmers.
+
+For implementation-oriented RFCs (e.g. for compiler internals), this section should focus on how compiler contributors should think about the change, and give examples of its concrete impact. For policy RFCs, this section should provide an example-driven introduction to the policy, and explain its impact in concrete terms.
+-->
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+## Item pages will be dependent only on the namespace
+
+Rust has three namespaces. For simplicity, this will only consider items that can be at the module level,
+since function locals cannot be documented.
+
+1. The value namespace. This includes `fn`, `const`, and `static`.
+2. The type namespace. This includes `mod`, `struct`, `union`, `enum`, `trait`, `type`, struct fields, and enum variants.
```

Comment:
```suggestion
2. The type namespace. This includes `struct`, `union`, `enum`, `trait`, and `type`.
```

* struct fields are not at the module level
* `doc(inline)` on enum variant re-exports does not work
* `mod`s have folders rather than files (this one I'm 50/50 on including in the list)

Reply:
Ah hmm, maybe I should move this to the section on URL fragments instead. I want to keep `mod` - I'll mention that it uses a folder instead of getting its own page.